webapp/channels/src/components/convert_gm_to_channel_modal/convert_gm_to_channel_modal.tsx

```diff
@@ -0,0 +1,220 @@
+import type {Team} from '@mattermost/types/teams';
+import type {UserProfile} from '@mattermost/types/users';
+
+import type {ActionResult} from 'mattermost-redux/types/actions';
+import {displayUsername} from 'mattermost-redux/utils/user_utils';
+
+import {getGroupMessageMembersCommonTeams} from 'actions/team_actions';
+import {trackEvent} from 'actions/telemetry_actions';
+
+import ChannelNameFormField from 'components/channel_name_form_field/channel_name_form_field';
+import type {Actions} from 'components/convert_gm_to_channel_modal/index';
+import NoCommonTeamsError from 'components/convert_gm_to_channel_modal/no_common_teams/no_common_teams';
+import TeamSelector from 'components/convert_gm_to_channel_modal/team_selector/team_selector';
+import WarningTextSection from 'components/convert_gm_to_channel_modal/warning_text_section/warning_text_section';
+import LoadingSpinner from 'components/widgets/loading/loading_spinner';
+
+export type Props = {
+    onExited: () => void;
+    channel: Channel;
+    actions: Actions;
+    profilesInChannel: UserProfile[];
+    teammateNameDisplaySetting: string;
+    channelsCategoryId: string | undefined;
+    currentUserId: string;
+}
+
+const ConvertGmToChannelModal = (props: Props) => {
+    const intl = useIntl();
+    const {formatMessage} = intl;
+
+    const [channelName, setChannelName] = useState<string>('');
+    const handleChannelNameChange = useCallback((newName: string) => {
+        setChannelName(newName);
+    }, []);
+
+    const [channelURL, setChannelChannelURL] = useState<string>('');
+    const handleChannelURLChange = useCallback((newURL: string) => {
+        setChannelChannelURL(newURL);
+    }, []);
+
+    const [channelMemberNames, setChannelMemberNames] = useState<string[]>([]);
+
+    useEffect(() => {
+        const validProfilesInChannel = props.profilesInChannel.
+            filter((user) => user.id !== props.currentUserId && user.delete_at === 0).
+            map((user) => displayUsername(user, props.teammateNameDisplaySetting));
+
+        setChannelMemberNames(validProfilesInChannel);
+    }, [props.profilesInChannel]);
+
+    const [commonTeamsById, setCommonTeamsById] = useState<{[id: string]: Team}>({});
+    const [commonTeamsFetched, setCommonTeamsFetched] = useState<boolean>(false);
+    const [loadingAnimationTimeout, setLoadingAnimationTimeout] = useState<boolean>(false);
+    const [selectedTeamId, setSelectedTeamId] = useState<string>();
+    const [nameError, setNameError] = useState<boolean>(false);
+    const [conversionError, setConversionError] = useState<string>();
+
+    const handleTeamChange = useCallback((teamId: string) => {
+        setSelectedTeamId(teamId);
+    }, []);
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        const work = async () => {
+            const response = await dispatch(getGroupMessageMembersCommonTeams(props.channel.id)) as ActionResult<Team[], ServerError>;
```

Comment:
This takes an arbitrary amount of time to get executed, so the component may be unmounted when you finish.

That is an issue because later you are calling `setX` functions, and that means update the state of an unmounted object.

I would use a `mounted` ref to avoid this kind of issues.

With a mounted ref I mean something like this:
```
const mounted = useRef(false);
useEffect(() => {
    mounted.current = true;
    return (() => {
        mounted.current = false;
    }
});
```

Reply:
Is that still needed with functional components and hooks? I feel like I'm mostly used to seeing that pattern in class components

Reply:
Done. Please verify if i'm doing it right.